### PR TITLE
Feature/ant plugin test

### PR DESCRIPTION
**Pull request by ZenHarbinger**
Updated the ant plugin to remove calls to self.run() for copy and mkdir -p
Use of shutil.copy() and os.makedirs.

Added ant test file.

https://bugs.launchpad.net/snapcraft/+bug/1600410

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

File: snapcraft/plugins/ant.py [hunk 1]
```diff
@@ -49,8 +50,9 @@ def build(self):
         if not files:
             raise RuntimeError('could not find any built jar files for part')
         jardir = os.path.join(self.installdir, 'jar')
-        self.run(['mkdir', '-p', jardir])
-        self.run(['cp', '-a'] + files + [jardir])
+        os.makedirs(jardir)
+        for f in files:
+            shutil.copy(f, jardir)
```
Comment (sergiusens) [inline snapcraft/plugins/ant.py]: may I suggest using `os.link` here?

**Comment by sergiusens** [comment]
Great conribution! I always like seeing tests!
I added a small minor comment, please tell me what you think, but it would reduce disk consumption if done.

**Comment by ZenHarbinger** [comment]
If os.link is that much better, sure. I'll push a change up in a second.

**Comment by ZenHarbinger** [comment]
Eh, I take it back, can we just copy since otherwise I have to get the file name so to create the link instead of just the target directory to copy to?

**Comment by sergiusens** [comment]
ok to test

**Pull request merged**